### PR TITLE
fix(49306): Missing Compiler API predicate functions: isImportTypeAssertionContainer

### LINKED ISSUE #49306: Missing Compiler API predicate functions: isImportTypeAssertionContainer
## Search Terms

isImportTypeAssertionContainer

## Suggestion

TypeScript's Compiler APIs include `is` functions for all kinds of Nodes. For example, `isClassDeclaration`, `isBigIntLiteral`, and so on.

In TypeScript 4.7 a new kind of node is introduced: `ImportTypeAssertionContainer`, but no `isImportTypeAssertionContainer` predicate function is exposed. For consistency with other kinds of Nodes, TypeScript should arguably also expose such a predicate function.

## Use Cases

- Custom Transformers (visiting and transforming Nodes in an AST)
- Working with the Compiler APIs
- etc

## Examples

```typescript
function visitNode (node: TS.Node, context: TS.TransformationContext, typescript: typeof TS): TS.VisitResult<TS.Node> {
    if (typescript.isImportTypeAssertionContainer(node) {
        return visitImportTypeAssertionContainer(node, context, typescript);
    }

    else if (...) {
        ....
    }
}
```

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #49306

### CODE REVIEW

File: tests/baselines/reference/api/tsserverlibrary.d.ts [hunk 1]
```diff
@@ -4724,6 +4724,7 @@ declare namespace ts {
     function isImportEqualsDeclaration(node: Node): node is ImportEqualsDeclaration;
     function isImportDeclaration(node: Node): node is ImportDeclaration;
     function isImportClause(node: Node): node is ImportClause;
+    function ImportTypeAssertionContainer(node: Node): node is ImportTypeAssertionContainer;
```
Comment (Andarist) [inline tests/baselines/reference/api/tsserverlibrary.d.ts:4727]:
```suggestion
    function isImportTypeAssertionContainer(node: Node): node is ImportTypeAssertionContainer;
```

File: src/compiler/factory/nodeTests.ts [hunk 2]
```diff
@@ -597,6 +597,10 @@ namespace ts {
         return node.kind === SyntaxKind.ImportClause;
     }
 
+    export function ImportTypeAssertionContainer(node: Node): node is ImportTypeAssertionContainer {
```
Comment (Andarist) [inline src/compiler/factory/nodeTests.ts:600]:
```suggestion
    export function isImportTypeAssertionContainer(node: Node): node is ImportTypeAssertionContainer {

```

File: tests/baselines/reference/api/typescript.d.ts [hunk 3]
```diff
@@ -4724,6 +4724,7 @@ declare namespace ts {
     function isImportEqualsDeclaration(node: Node): node is ImportEqualsDeclaration;
     function isImportDeclaration(node: Node): node is ImportDeclaration;
     function isImportClause(node: Node): node is ImportClause;
+    function ImportTypeAssertionContainer(node: Node): node is ImportTypeAssertionContainer;
```
Comment (Andarist) [inline tests/baselines/reference/api/typescript.d.ts:4727]:
```suggestion
    function isImportTypeAssertionContainer(node: Node): node is ImportTypeAssertionContainer;
```

**Comment by DanielRosenwasser** [comment]
Should we be adding this if technically `resolution-mode` on an `import(...)` type should be an error?

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-4.7

**Pull request merged**